packages/integrations/components/pico/observer.js

```diff
@@ -0,0 +1,200 @@
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import debounce from 'lodash/debounce';
+import useMutationObserver from '../../hooks/useMutationObserver';
+import {
+  actionVerifyPicoUser,
+  actionRequireUpgrade,
+  actionReceivePicoPlanUpgrade,
+} from '../../actions/picoActions';
+import {
+  showUpgradeModalSelector,
+  upgradeModalDismissedSelector,
+} from '../../selectors/coralSelector';
+
+const PicoObserver = ({
+  tiers,
+  accessToken,
+  logoutRequestAction,
+}) => {
+  // Declare state variables for effects to be run against.
+  const [email, setEmail] = useState(undefined);
+  const [isAnonymous, setIsAnonymous] = useState(false);
+  const [isRegistered, setIsRegistered] = useState(false);
+  const [isPaying, setIsPaying] = useState(false);
+  const [canComment, setCanComment] = useState(false);
+
+  const manageLocalState = (target) => {
+    const emailAddress = target.getAttribute('data-pico-email');
+    const status = target.getAttribute('data-pico-status');
+    const tier = target.getAttribute('data-pico-tier');
+
+    const hasValidStatus = 'registered' === status || 'paying' === status;
+    const hasNotUpdated = ! isAnonymous || ! isRegistered || ! isPaying;
+
+    if ('excluded' !== status && hasValidStatus && hasNotUpdated) {
+      // Update the `email` value with the user's email address.
+      if (undefined === email) {
+        setEmail(emailAddress);
+      }
+      // If the user's tier is included in the whitelisted tiers for SSO,
```

Comment:
Avoid use of "whitelist" in favor of more inclusive language.

```suggestion
      // If the user's tier is included in the allowed list of tiers for SSO,
```